### ISSUE TITLE
Open3.popen2e and Open3.popen2 do not behave as expected

### ISSUE THREAD

**Issue by Seanny123** [jruby/jruby#1045]
I can successfully open a pipe with the following code in Ruby 1.9.3:

```
require "open3"
ain, aout, wait = Open3.popen2e("echo 'hello'")
```

But when I run the same code with JRuby 1.7.4, I get the following result:

```
require "open3"
ain, aout, wait = Open3.popen2e("echo 'hello'")
```

```
file:/home/user/.rvm/rubies/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/process.rb:4 warning: unsupported spawn option: in
TypeError: wrong exec option: [:out, :err]
from org/jruby/RubyProcess.java:1067:in `_spawn_internal'
```

`Open3.popen2` also give a weird result, that is not necessarily a bug, but might give you a better idea might be going wrong:

```
require "open3"
ain, aout, wait = Open3.popen2("echo 'hello'")
```

```
file:/home/user/.rvm/rubies/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/process.rb:4 warning: unsupported spawn option: in
file:/home/user/.rvm/rubies/jruby-1.7.4/lib/jruby.jar!/jruby/kernel19/process.rb:4 warning: unsupported spawn option: out
```

However `Open3.popen3` does work as expected. My OS is Ubuntu 12.04.

**Comment by headius** [comment]
This should work just fine on JRuby 9k, which has a new truly-native spawn. For JRuby 1.7, it may just require hacking our modified open3.rb to support another form of popen3, which *does* work. I'll have a look.

**Comment by ripienaar** [comment]
I tested this on 9k and it works for me

```
Open3.popen2e("test.sh") do |stdin, out, wait_thr|
  stdin.close

  puts out.gets until out.eof?
end
```

does what you'd expect

**Comment by headius** [comment]
@ripienaar Thanks for the confirmation.

More work is needed here to get open2e and the other forms to work right.

**Comment by enebo** [comment]
This works in 9k and we will never fix this in 1.7.x as it is winding down....resolving.
